src/padding.rs

```diff
@@ -1,12 +1,130 @@
-// Applies new padding to a BSON packet.
-// Returns the padded packet.
+use rand::Rng;
+use std::io::{Error, ErrorKind};
+
+/// Generates a vector of random bytes with the specified length, excluding a specific character.
+/// # Arguments
+/// * `rng` - A mutable reference to a random number generator.
+/// * `count` - The number of bytes to generate.
+/// * `exclusion` - The character to exclude from the generated bytes.
+/// # Returns
+/// A vector of random bytes with the defined length, excluding the specified character.
+fn generate_random_bytes<R: Rng>(rng: &mut R, count: u32, exclusion: char) -> Vec<u8> {
```

Comment:
This can probably be inlined, it can be done fairly simply. Generics here really aren't necessary and this solution is overkill, especially since we are drawing 256 even if we only need 2 (which on low entropy systems is a huge issue perf wise)